apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/ApolloRequest.kt

```diff
@@ -23,8 +23,12 @@ private constructor(
     override val canBeBatched: Boolean?,
 ) : ExecutionOptions {
 
-  fun newBuilder(): Builder<D> {
-    return Builder(operation)
+  fun newBuilder(): Builder<D> = duplicate(Builder(operation))
+
+  fun copy(operation: Operation<D>): ApolloRequest<D> = duplicate(Builder(operation)).build()
```

Comment:
Thanks for sending this ! I understand the use case but I'd like to avoid the "copy" vocabulary here because it hints heavily at data classes which are not really used here.

Instead, maybe use `newBuilder(operation: Operation<E>)`? It also makes it more discoverable?

```kotlin
  fun newBuilder(): Builder<D> {
    return newBuilder(operation)
  }

  fun <E: Operation.Data> newBuilder(operation: Operation<E>): Builder<E> {
```